bpf/bpf_overlay.c

```diff
@@ -458,23 +465,36 @@ static __always_inline int handle_ipv4(struct __ctx_buff *ctx,
 	}
 #endif /* ENABLE_EGRESS_GATEWAY_COMMON */
 
-	/* Deliver to local (non-host) endpoint: */
-	ep = lookup_ip4_endpoint(ip4);
-	if (ep && !(ep->flags & ENDPOINT_MASK_HOST_DELIVERY))
-		return ipv4_local_delivery(ctx, ETH_HLEN, *identity, MARK_MAGIC_IDENTITY,
-					   ip4, ep, METRIC_INGRESS, false, true, 0);
+#if defined(ENABLE_DSR) && defined(IS_BPF_OVERLAY) && (DSR_ENCAP_MODE == DSR_ENCAP_GENEVE)
```

Comment:
```suggestion
#if defined(ENABLE_DSR) && (DSR_ENCAP_MODE == DSR_ENCAP_GENEVE)
```

We're in `bpf_overlay`, no need to check further :).

I suspect you could even completely remove these guards, and rely on `is_dsr` being optimized out when DSR is not enabled. But that's just a hunch.